pwiz_tools/Skyline/Model/ModificationMatcher.cs

```diff
@@ -146,6 +146,31 @@ private IEnumerable<AAModInfo> EnumerateSequenceInfos(string seq, bool includeUn
             bool isSpecificHeavy = FastaSequence.OPEN_MOD.All(paren => aas.Length > seq.Count(c => c == paren));
             var lossOnlyMods = Settings.PeptideSettings.Modifications.StaticModifications
                 .Where(m =>m.HasLoss && !m.HasMod).ToArray();
+            foreach (var modInfo in EnumerateMods(seq, false))

```

Comment:
I think I need you to walk me through the need for such big changes to ModificationMatcher.

Reply:
These changes are in order to share the modification enumeration code between ModificationMatcher and the new ModifiedSequence ctor (that takes a modified sequence instead of an unmodified one and a list of mods). Happy to have a meeting to go over it though.